modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/internal/type/test/EmailNotificationTypeTest.java

```diff
@@ -468,18 +521,39 @@ public void testSendNotificationWithAccountRoles() throws Exception {
 		// Send email with an object definition not restricted by account entry
 
 		_testSendNotificationWithAccountRoles(
-			null, user2.getEmailAddress(), 1,
+			null,
+			StringUtil.merge(
+				ListUtil.fromArray(
+					user2.getEmailAddress(), user5.getEmailAddress())),
+			1,
 			StringUtil.merge(
 				ListUtil.fromArray(
 					user1.getEmailAddress(), user2.getEmailAddress(),
-					user3.getEmailAddress())),
+					user3.getEmailAddress(), user4.getEmailAddress())),
 			notificationTemplate);
 
 		// Send email with an object definition restricted by account entry
 
+		_userGroupRoleLocalService.addUserGroupRole(
+			user4.getUserId(), organization1.getGroupId(),
+			organizationRole2.getRoleId());
+
+		_accountEntryOrganizationRelLocalService.addAccountEntryOrganizationRel(
+			accountEntry3.getAccountEntryId(),
+			organization2.getOrganizationId());
+
+		User user6 = UserTestUtil.addUser();
+
+		_userGroupRoleLocalService.addUserGroupRole(
+			user6.getUserId(), parentOrganization.getGroupId(),
+			organizationRole2.getRoleId());
+
 		_testSendNotificationWithAccountRoles(
```

Comment:
@devpedrotavares please rename this private method to `_testSendNotificationWithRoles`